src/main/java/com/aws/greengrass/mqttbridge/util/BatchedSubscriber.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.mqttbridge.util;
+
+import com.aws.greengrass.config.ChildChanged;
+import com.aws.greengrass.config.Node;
+import com.aws.greengrass.config.Subscriber;
+import com.aws.greengrass.config.Topic;
+import com.aws.greengrass.config.Topics;
+import com.aws.greengrass.config.WhatHappened;
+import lombok.NonNull;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.BiConsumer;
+import java.util.function.BiPredicate;
+
+/**
+ * NOTE: This class is copied from nucleus, so that we can maintain compatibility with older versions
```

Comment:
Can this live in Nucleus actually? What is the customization here?

Reply:
I just readjusted the wording to hopefully make it a bit clearer 👍